autox/config.go

```diff
@@ -7,89 +7,73 @@ import (
 	"path/filepath"
 
 	"github.com/iter8-tools/iter8/base/log"
-	"k8s.io/apimachinery/pkg/runtime/schema"
 
 	"sigs.k8s.io/yaml"
 )
 
-// resourceConfig is the configuration that identifies the resources to watch and in which namespaces.
-type resourceConfig struct {
-	// Namespaces is list of namespaces to watch
-	Namespaces []string `json:"namespaces,omitempty" yaml:"namespaces,omitempty"`
+// trigger specifies when a releaseGroupSpec should be installed
+type trigger struct {
+	Group string `json:"group,omitempty" yaml:"group,omitempty"`
 
-	// Resources is list of resoure types that should be watched in the namespaces
-	Resources []schema.GroupVersionResource `json:"resources,omitempty" yaml:"resources,omitempty"`
-}
+	Version string `json:"version,omitempty" yaml:"version,omitempty"`
 
-// chart points to a particular Helm chart
-type chart struct {
-	// Repo is the repo of the Helm chart
-	Repo string `json:"repo" yaml:"repo"`
+	Resource string `json:"resource,omitempty" yaml:"resource,omitempty"`
 
-	// Name is the name of the Helm chart
-	Name string `json:"name" yaml:"name"`
+	Name string `json:"name,omitempty" yaml:"name,omitempty"`
 
-	// ValuesFileURL is the URL to the values file of the Helm chart
-	ValuesFileURL string `json:"valuesFileURL" yaml:"valuesFileURL"`
+	Namespace string `json:"namespace,omitempty" yaml:"namespace,omitempty"`
 }
 
-// chartGroup is the configuration of all the Helm charts for a particular experiment group
-// the key is an identifier for the chart
-type chartGroup struct {
-	Charts map[string]chart `json:"charts" yaml:"charts"`
-}
+// releaseSpec points to a particular Helm releaseSpec
+type releaseSpec struct {
+	// RepoURL is the url of the Helm repo
+	RepoURL string `json:"repoURL" yaml:"repoURL"`
```

Comment:
Please drop the `repoURL string` altogether in the releaseSpec **for now**.

We want to hardcode the repoURL **for now** in all the releaseSpecs (as the Iter8 Hub URL, instead of making this configurable by the user). Mainly for security reasons.